csharp/ql/src/semmle/code/csharp/dataflow/internal/rangeanalysis/SignAnalysisSpecific.qll

```diff
@@ -81,6 +81,9 @@ predicate unknownIntegerAccess(Expr e) {
   or
   // unchecked
   e instanceof CS::UncheckedExpr and e.getType() instanceof NumericOrCharType
+  or
+  // await
+  e instanceof CS::AwaitExpr and e.getType() instanceof NumericOrCharType
```

Comment:
I would also expect this to be defined in `specificSubExprSign`.

Reply:
Await is behaving similarly to a `Call`. We have now clue about what the sign can be, so I think it belongs to `unknownIntegerAccess`. Why do you think it should go to `specificSubExprSign`?

Reply:
If we at some point have inter-procedural sign analysis, it should also work for `async` methods. Adding `result = exprSign(e.(AwaitExpr).getExpr())` to `specificSubExprSign` will make sure that we get that.

Reply:
This doesn't seem to work. The type of the underlying expression is probably `Task<int>`, and therefore has no sign.

Reply:
Ah yes.